pkg/sql/catalog/lease/descriptor_version_state.go

```diff
@@ -65,6 +67,9 @@ type descriptorVersionState struct {
 		// when the version isn't associated with a lease.
 		expiration hlc.Timestamp
 
+		// The session that was used to acquire this descriptor version.
+		session sqlliveness.Session
```

Comment:
I see a lot of `nil` checks across this struct. When is `session` expected to be nil vs not? Can that be noted on the field itself?